File: evolutionSimulation/c++/animals/create.cpp
#include <vector>
#include <iostream>
#include <iomanip>
#include <string>
#include <vector>

using namespace std; 

class Animals {
    public: 
        string species;
        string name;
        string brain;
        int id;
        int status;
        int mating;
        int speed; 
        int eyesight;
        int fov;
};

Animals animal_placeholder;

Animals matrix[3][4]  = {
    {animal_placeholder, animal_placeholder, animal_placeholder, animal_placeholder}, 
    {animal_placeholder, animal_placeholder, animal_placeholder, animal_placeholder}, 
    {animal_placeholder, animal_placeholder, animal_placeholder, animal_placeholder}, 
};

vector <Animals> list;

int main() {
    for (int i = 0; i <= 11; i++) {
        Animals temp;
        temp.name = "sheep" + to_string(i);
        temp.id = i;
        list.push_back(temp);
        int row = i / 4;
        int column = i % 4;
        matrix[row][column] = temp;
    }

    for (int row = 0; row <= 2; row ++ ) {
        for (int column = 0; column <= 3; column++) {
            Animals temp = matrix[row][column];
            cout << temp.name;
        }
        cout << endl;
    }
    
    //cout << sheep.name;
    return 0;;
}

File: evolutionSimulation/python/animals/create.py
from evolutionSimulation.python.neuralnetworks.nn import * 
from datasets import load_dataset
import torch
import transformers
import time
import torch.nn as nn 
import torch.nn.functional as F
import torch.optim as optim
import random


DEVICE = torch.device("cuda" if torch.cuda.is_available() else "cpu")

class Animals():
    def __init__(self, species, name, brain_path, num, stats, status, mating, speed, eyesight, fov):
            brain = Brain()
            self.species = species
            self.name = name    
            if brain_path: 
                self.brain = brain.load_state_dict(torch.load(brain_path)).to(DEVICE)
            else:
                self.brain = brain
            brain.num = num
            brain.stats = stats
            brain.status = status
            brain.mating = mating
            brain.speed = speed 
            brain.eyesight = eyesight 
            brain.fov = fov
